packages/dialtone-vue3/common/emoji/index.js

```diff
@@ -1,6 +1,7 @@
-import emojiRegex from 'emoji-regex';
+import { emojiPattern } from 'regex-combined-emojis';
```

Comment:
You're right, it doesn't have many weekly downloads and it's bigger in size.

The reason I choose that library instead of `emoji-regex` was due to this:

> Note: because the regular expression **has the global flag set**, this module
> exports a function that returns the regex rather than exporting the regular
> expression itself, **to make it impossible to** (accidentally) **mutate the
> original regular expression**.

We can use that regular expression to detect on paste, but it cannot be used to detect on user input (selecting an emoji from the emoji selector) because we need to remove the global flag an append the `$` to match the end of the string.

Also the `emoji-regex` library was published 7 moths ago.

![Screenshot 2024-05-27 at 11 27 13 a m](https://github.com/dialpad/dialtone/assets/87546543/96b1f516-654e-4c97-b1d0-da08f3bd4d1e)

And the `regex-combined-emojis` its being updated more regularly.

![Screenshot 2024-05-27 at 11 27 35 a m](https://github.com/dialpad/dialtone/assets/87546543/6226bf1e-5895-480e-b19c-bd36da7de789)

Reply:
The emoji-regex library only needs to update when the Unicode standard for emojis updates. The most recent update to unicode was September 2023 with Unicode version 15.1, so they haven't had to publish since then.

Reply:
> We can use that regular expression to detect on paste, but it cannot be used to detect on user input (selecting an emoji from the emoji selector) because we need to remove the global flag an append the `$` to match the end of the string.

Hmm maybe we just pull ourselves from https://www.npmjs.com/package/emoji-test-regex-pattern. This is where emoji-regex gets it's data from. Then we can modify ourselves as we need. see `emoji-test-regex-pattern/dist/latest/javascript.txt`

Reply:
> > We can use that regular expression to detect on paste, but it cannot be used to detect on user input (selecting an emoji from the emoji selector) because we need to remove the global flag an append the `$` to match the end of the string.
> 
> Hmm maybe we just pull ourselves from https://www.npmjs.com/package/emoji-test-regex-pattern. This is where emoji-regex gets it's data from. Then we can modify ourselves as we need. see `emoji-test-regex-pattern/dist/latest/javascript.txt`

What's the difference of using that instead of `regex-combined-emojis`? It is even bigger in size and it'd need more work to parse from `javascript.txt` file and create the regex from the output.

Reply:
Yeah guess its fine 🤷‍♂️